api/dao/basecontainerstorage.py

```diff
@@ -124,6 +130,70 @@ def get_children(self, _id, projection=None, uid=None):
             projection = {'info': 0, 'files.info': 0, 'subject': 0, 'tags': 0}
         return ContainerStorage.factory(child_name).get_all_el(query, None, projection)
 
+
+    def get_children(self, _id, projection=None, uid=None):
+        child_name = self.child_cont_name
+        if not child_name:
+            raise APIStorageException('Children cannot be listed from the {0} level'.format(self.cont_name))
+        if not self.use_object_id:
+            query = {containerutil.singularize(self.cont_name): _id}
+        else:
+            query = {containerutil.singularize(self.cont_name): bson.ObjectId(_id)}
+
+        if uid:
+            query['permissions'] = {'$elemMatch': {'_id': uid}}
+        if not projection:
+            projection = {'info': 0, 'files.info': 0, 'subject': 0, 'tags': 0}
+        return ContainerStorage.factory(child_name).get_all_el(query, None, projection)
+
+
+    def get_parent_tree(self, _id, cont=None, projection=None, add_self=False):
```

Comment:
Are there much better ways I could have done this tree walking? Absolutely. 

I plan on rewriting it to be a bit more recursive when we formalize subjects, currently scheduled as the next big project.